pytorch_lightning/core/hooks.py

```diff
@@ -805,6 +813,18 @@ def on_after_batch_transfer(self, batch, dataloader_idx):
         """
         return batch
 
+    @property
+    def prepare_data_per_node(self) -> bool:
+        """
+        If True, each LOCAL_RANK=0 will call prepare data.
+        Otherwise only NODE_RANK=0, LOCAL_RANK=0 will prepare data.
+        """
+        return self._prepare_data_per_node
+
+    @prepare_data_per_node.setter
+    def prepare_data_per_node(self, prepare_data_per_node: bool) -> None:
+        self._prepare_data_per_node = prepare_data_per_node
+
```

Comment:
let's also add `__jit_unused_properties__` to this class so that the property isn't included in the torchscript export
and we can add this exclusion list to the lightning module as below:

https://github.com/PyTorchLightning/pytorch-lightning/blob/f87b2ef21f01d7bbda666088b8cdae3dcbe12010/pytorch_lightning/core/lightning.py#L65-L83